### PR TITLE
serial mode, fixed II

### PR DESCRIPTION
**Pull request by nhanvtran**
This makes a very simplistic serial mode work.  Also explicitly declares the II inside of nn_layer.h to fix any weird pipeline optimizations.

do i need to add this fix to nn_conv.h and also any changes needed for PyTorch?

### CODE REVIEW

File: keras-to-hls/keras-config.yml [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
-OutputDir: my-hls-dir-test
+KerasJson: example-keras-model-files/KERAS_1layer.json
+KerasH5:   example-keras-model-files/KERAS_1layer_weights.h5
+OutputDir: my-hls-test
 ProjectName: myproject
 XilinxPart: xc7vx690tffg1927-2
 ClockPeriod: 5
 
-IOType: io_parallel # options: io_serial/io_parallel
+IOType: io_parallelv # options: io_serial/io_parallel
```
Comment (benjaminkreis) [inline keras-to-hls/keras-config.yml]: typo?  parallelv

**Comment by benjaminkreis** [comment]
PS your changes are in hls_writer which is used by both the keras and pytorch converter, so this should be all that's required

**Comment by nhanvtran** [comment]
@benjaminkreis yes, I just keep the reshapes and partitions for now.  I don't think there's anything wrong with it particularly.  However, I do see that the partitioned arrays are still stored in BRAMs to use the resources efficiently.  

thanks for picking up on the typo -- it's fixed now.

for the conv layer, we need to add the configuration which asks for io_type

**Comment by benjaminkreis** [comment]
This looks okay to me, at least as a first version.  As you know, the IP expects all inputs on one clock edge, so if the data is streaming in, the user would have to add a wrapper around it.  The long term goal could be to recover the streaming functionality, but no reason not to merge this now in my opinion.

**Comment by ejk43** [comment]
Hey guys, take a look here as a follow-up PR: https://github.com/hls-fpga-machine-learning/hls4ml/pull/48

I think I've hit a good mix of directives that gets back towards the serialized behavior I hit last year in the dense layer.

**Pull request merged**